estate/models/estate_property.py

```diff
@@ -1,18 +1,45 @@
+
+#python import
 from email.policy import default
+from dateutil.relativedelta import relativedelta
 
+#odoo import
 from odoo import models, fields
 
+
+def _date_in_three_months(self:None):
```

Comment:
On utilise pas le typing en Odoo, et si tu veux l'utiliser alors il faut utiliser le type approprié: self est un recordset non null
```suggestion
def _date_in_three_months(self):
```